File: src/main/java/com/example/ajaxonboardingproject/model/Role.java
package com.example.ajaxonboardingproject.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@Table(name = "roles")
public class Role {
    @Id
    @EqualsAndHashCode.Exclude
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    @Enumerated(value = EnumType.STRING)
    private RoleName name;

    public enum RoleName {
        ADMIN, USER
    }
}

File: src/main/java/com/example/ajaxonboardingproject/model/Role.kt
package com.example.ajaxonboardingproject.model

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document("roles")
data class Role(
//    @Column(unique = true)
    @JsonProperty("name")
//    @Enumerated(value = EnumType.STRING)
    var name: RoleName,
) {

    enum class RoleName {
        ADMIN, USER
    }
}

